temporal-sdk/src/main/java/io/temporal/internal/sync/POJOActivityTaskHandler.java

```diff
@@ -287,6 +301,96 @@ public Object execute(Object[] arguments) {
     }
   }
 
+  private class DynamicActivityImplementation implements ActivityTaskExecutor {
+
+    private final DynamicActivity activity;
+
+    public DynamicActivityImplementation(DynamicActivity activity) {
+      this.activity = activity;
+    }
+
+    @Override
+    public ActivityTaskHandler.Result execute(ActivityInfoImpl info, Scope metricsScope) {
```

Comment:
Quite a lot of the code in here is quite similar to the above `POJOActivityImplementation` -- would be nice to dedupe as reasonable.